git-sync-relay/gitsync.sh

```diff
@@ -11,7 +11,8 @@ fi
 
 if [ -n "$(ls -A "$GIT_SYNC_ROOT")" ]; then
   echo "Error: GIT_SYNC_ROOT ${GIT_SYNC_ROOT} is not empty."
-  exit 1
+  echo "cleaning the directory..."
+  rm -rf "$GIT_SYNC_ROOT"
```

Comment:
Yeah, I'm really curious how this could be not empty if it's from an emptyDir.

Reply:
slack thread: <https://astronomer.slack.com/archives/C04JS0GADTL/p1686296023765619?thread_ts=1686123723.437749&cid=C04JS0GADTL>

my understanding of the problem is this:

> if i understand this correctly, the git-sync container restarts, but the pod does not, which causes the old state to exist in the pod volume, which causes the git-sync container to fail when it starts up any time after the first time

if that's the case, the better fix would be to detect that condition and use the old state if it's good. imagine a case where a git repo is several gigabytes large (sad, but this does exist). reusing that existing state would be really beneficial rather than downloading all that data again.